src/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -269,6 +271,19 @@ public SamReader open(final SamInputResource resource) {
                             bufferedStream.close();
                             primitiveSamReader = new CRAMFileReader(sourceFile, indexFile, referenceSource, validationStringency);
                         }
+                    } else if (SamStreams.isSRAFile(bufferedStream)) {
+                        if (sourceFile == null || !sourceFile.isFile()) {
+                            sourceFile = null;
```

Comment:
If `sourceFile` is null, it appears that you'll get a `NullPointerException` in the `SRAFileReader` construction below. If you can't handle piped input from an SRA file, throw an exception saying so.